File: src/main/java/investLee/platform/ecommerce/domain/entity/ProductOptionGroupEntity.java
package investLee.platform.ecommerce.domain.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "product_option_groups")
@Getter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ProductOptionGroupEntity {

    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;

    private int displayOrder;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "product_id")
    private ProductEntity product;

    @OneToMany(mappedBy = "optionGroup", cascade = CascadeType.ALL, orphanRemoval = true)
    @Builder.Default
    private List<ProductOptionEntity> options = new ArrayList<>();
}

File: src/infrastructure/entities/Product_Option_Group.entity.ts
import { Entity, PrimaryGeneratedColumn, Column, ManyToOne } from "typeorm";

import ProductEntity from "./Product.entity";

@Entity("product_option_groups")
export default class ProductOptionGroupEntity {
  @PrimaryGeneratedColumn("increment")
  id: number;

  @ManyToOne(() => ProductEntity, { onDelete: "CASCADE" })
  product: ProductEntity;

  @Column({ type: "varchar", length: 100 })
  name: string;

  @Column({ type: "int", default: 0 })
  display_order: number;
}
